From f14fa33b2c5452be7df163973c26a0d1222696a2 Mon Sep 17 00:00:00 2001
From: Alvin Wong <alvin@alvinhc.com>
Date: Thu, 18 Apr 2019 18:59:58 +0800
Subject: [PATCH 17/17] Hack to hide 1px border with OpenGL fullscreen hack

Unfortunately can't hide all four sides because the bug returns. Now we
leave the bottom border visible, which is probably the most we can do.

Ref: https://bugreports.qt.io/browse/QTBUG-41309
---
 src/plugins/platforms/windows/qwindowswindow.cpp | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/src/plugins/platforms/windows/qwindowswindow.cpp b/src/plugins/platforms/windows/qwindowswindow.cpp
index 6949b9c669..10de52d4d6 100644
--- a/src/plugins/platforms/windows/qwindowswindow.cpp
+++ b/src/plugins/platforms/windows/qwindowswindow.cpp
@@ -1634,7 +1634,7 @@ void QWindowsWindow::show_sys() const
                 restoreMaximize = true;
         } else {
             updateTransientParent();
-            if (state & Qt::WindowMaximized) {
+            if (state & Qt::WindowMaximized && !(state & Qt::WindowFullScreen)) {
                 sm = SW_SHOWMAXIMIZED;
                 // Windows will not behave correctly when we try to maximize a window which does not
                 // have minimize nor maximize buttons in the window frame. Windows would then ignore
@@ -2124,7 +2124,7 @@ bool QWindowsWindow::isFullScreen_sys() const
         return false;
     QRect geometry = geometry_sys();
     if (testFlag(HasBorderInFullScreen))
-        geometry += QMargins(1, 1, 1, 1);
+        geometry += QMargins(0, 0, 0, 1);
     QPlatformScreen *screen = screenForGeometry(geometry);
     return screen && geometry == screen->geometry();
 }
@@ -2195,7 +2195,11 @@ void QWindowsWindow::setWindowState_sys(Qt::WindowStates newState)
                 const UINT swpf = SWP_FRAMECHANGED | SWP_NOACTIVATE;
                 const bool wasSync = testFlag(SynchronousGeometryChangeEvent);
                 setFlag(SynchronousGeometryChangeEvent);
-                SetWindowPos(m_data.hwnd, HWND_TOP, r.left(), r.top(), r.width(), r.height(), swpf);
+                if (testFlag(HasBorderInFullScreen)) {
+                    SetWindowPos(m_data.hwnd, HWND_TOP, r.left() - 1, r.top() - 1, r.width() + 2, r.height() + 1, swpf);
+                } else {
+                    SetWindowPos(m_data.hwnd, HWND_TOP, r.left(), r.top(), r.width(), r.height(), swpf);
+                }
                 if (!wasSync)
                     clearFlag(SynchronousGeometryChangeEvent);
                 clearFlag(MaximizeToFullScreen);
-- 
2.20.1.windows.1

